webapp/src/components/boardTemplateSelector/boardTemplateSelectorItem.tsx

```diff
@@ -38,7 +39,9 @@ const BoardTemplateSelectorItem = (props: Props) => {
         >
             <span className='template-icon'>{template.icon}</span>
             <span className='template-name'>{template.title}</span>
-            {!template.templateVersion &&
```

Comment:
Replacing `template.templateVersion` with `template.teamId` check to fix a case where you create a board from a default template then convert that board again to a template. Template version stays the same all across, so we end up with edit and delete options hidden incorrectly from this template.